src/main.ts

```diff
@@ -13,17 +13,24 @@ if (selection.length !== 1 || !isShape(node)) {
   figma.closePlugin();
 }
 
-let fillIndex = getFillIndexWithGradient(node);
-if (fillIndex === null) {
+if (getFillIndexWithGradient(node) === null) {
   figma.notify("Please select a shape with a gradient");
   figma.closePlugin();
 }
 
-const fill = node.fills[fillIndex];
-console.log("Fill:", fill);
+let fills = node.fills as readonly Paint[];
 
-const newColors = easeGradient(fill, 15);
-console.log("New colors:", newColors);
+fills.forEach((fill, fillIndex) => {
+  if (isGradient(fill)) {
+    console.log("Fill:", fill);
 
-applyColors(node, fillIndex, newColors);
-figma.closePlugin();
+    const newColors = easeGradient(fill, 15);
+    console.log("New colors:", newColors);
+
+    applyColors(node, fillIndex, newColors);
+  }
+
+  if (fillIndex >= fills.length - 1) {
+    figma.closePlugin();
+  }
+});
```

Comment:
It may be simpler to instead call `figma.closePlugin()` outside of the forEach:
```suggestion
});

figma.closePlugin();
```